### PR TITLE
Small fixed error in code

### PR DESCRIPTION
**Pull request by DarkBullNull**
If define CALL_ARG_STATS = 1 and DISPLAY_SIZES = 1 in "jit.h", then will be compilation errors 

- Remove unused pointers
- Fixed bool-check error call
- Fixed the type declaration for dataSize
- Grammar fix in "compiler.h"

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=51821) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=51821) before we can accept your contribution.<br/><br/>:x: DarkBullNull [sign now](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=51821)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/runtime?pullRequest=51821) it.</sub>

File: src/coreclr/jit/codegen.h [hunk 1]
```diff
@@ -208,6 +208,7 @@ class CodeGen final : public CodeGenInterface
     void**    codePtr;
     uint32_t* nativeSizeOfCode;
     unsigned  codeSize;
+    size_t    dataSize;
```
Comment (SingleAccretion) [inline src/coreclr/jit/codegen.h:211]: This should probably live with the other related fields in [compiler.h](https://github.com/dotnet/runtime/blob/79d8dadf915d7d13b0aee3933c5b3d51a4940159/src/coreclr/jit/compiler.h#L11226).

Comment (DarkBullNull) [reply src/coreclr/jit/codegen.h]: Hm, okay, i moved define dataSize in compiler.h
Thank you!

**Comment by DarkBullNull** [comment]
@BruceForstall

File: src/coreclr/jit/gentree.cpp [hunk 2]
```diff
@@ -16564,7 +16564,7 @@ bool GenTree::IsPhiDefn()
 //    comp      - the Compiler object.
 //
 // Return Value:
-//    Returns "true" iff 'this' is a GT_LCL_FLD or GT_STORE_LCL_FLD on which the type
+//    Returns "true" if 'this' is a GT_LCL_FLD or GT_STORE_LCL_FLD on which the type
```
Comment (sandreenko) [inline src/coreclr/jit/gentree.cpp:16567]: iff is widely used in the jit code base as "if and only if"

File: src/coreclr/jit/compiler.cpp [hunk 3]
```diff
@@ -7451,8 +7452,8 @@ void Compiler::compJitStats()
 
 void Compiler::compCallArgStats()
 {
-    GenTree* args;
-    GenTree* argx;
+    GenTree* args = nullptr;
```
Comment (sandreenko) [inline src/coreclr/jit/compiler.cpp:7455]: where is 'args' and 'argx' used in this method? Looks like they can be just deleted.

**Review by sandreenko — COMMENTED** [review]
> Fixed the type declaration for dataSize(because of 2523-2524 line doesn't see the declaration of this variable in "codegencommon.cpp"). Correct me, if i'm wrong

Is there an issue if we move 
```
size_t dataSize = GetEmitter()->emitDataSize();
```
from `genEmitMachineCode` to the next function instead of declaring it globally?

and revert eDataSize after.

File: src/coreclr/jit/compiler.cpp (hunk 3 above)
Comment (DarkBullNull) [reply src/coreclr/jit/compiler.cpp]: Thank you for reviewing! I've fixed this all.

File: src/coreclr/jit/codegencommon.cpp [hunk 4]
```diff
@@ -2377,11 +2377,7 @@ void CodeGen::genEmitMachineCode()
 
     compiler->unwindReserve();
 
-#if DISPLAY_SIZES
```
Comment (sandreenko) [inline src/coreclr/jit/codegencommon.cpp:2380]: the format job does not like these 3 empty lines, could you please delete 2?
You can fix it manually or apply the patch produced by the ci job (https://dev.azure.com/dnceng/_apis/resources/Containers/6976659/format.windows.x64.patch?itemPath=format.windows.x64.patch%2Fformat.patch) or run the format job locally ([doc](https://github.com/dotnet/jitutils/blob/main/doc/formatting.md)):
1. download https://github.com/dotnet/jitutils;
2. run bootstrap;
3. run `jit-format.bat --fix --coreclr path_to_runtime\runtime\src\coreclr`

**Review by sandreenko — APPROVED** [review]
LGTM, thank you for restoring `CALL_ARG_STATS` and `DISPLAY_SIZES `.

**Comment by sandreenko** [comment]
cc @dotnet/jit-contrib

File: src/coreclr/jit/codegencommon.cpp (hunk 4 above)
Comment (DarkBullNull) [reply src/coreclr/jit/codegencommon.cpp]: Thank you for your comment. Fixed it!

**Comment by sandreenko** [comment]
The failures are unrelated, covered by https://statics.teams.cdn.office.net/evergreen-assets/safelinks/1/atp-safelinks.html

**Pull request merged**